library/src/main/java/com/datatorrent/lib/formatter/Formatter.java

```diff
@@ -65,17 +73,28 @@ public void setup(PortContext context)
     @Override
     public void process(Object inputTuple)
     {
+      incomingTuplesCount++;
       OUTPUT tuple = convert(inputTuple);
       if (tuple == null && err.isConnected()) {
+        errorTupleCount++;
         err.emit(inputTuple);
         return;
       }
       if (out.isConnected()) {
+        emittedObjectCount++;
         out.emit(tuple);
       }
     }
   };
 
+  @Override
+  public void beginWindow(long windowId)
+  {
+    errorTupleCount = 0;
+    emittedObjectCount = 0;
+    incomingTuplesCount = 0;
+  }
+
```

Comment:
I think this is correct. Reset should be done in begin window only. These counts are picked between endWindow and beginWindow.. If reset is done in endWindow, then count will be 0 always.